dl/src/services/WalletService.js

```diff
@@ -117,47 +117,63 @@ class WalletService {
      * @param {String} accountName
      * @param {String} password
      */
-    static createWalletByAccount(accountName, password) {
+    static createWalletByAccount(account, password) {
 
         if(typeof password !== 'string') {
             throw new Error("password string is required");
         }
 
-
-        if(typeof accountName !== "string") {
-            throw new Error("accountName must be a string");
-        }
-
-        if(accountName.trim() === "") {
-            throw new Error("accountName can not be an empty string");
-        }
-
         let passwordAes = Aes.fromSeed(password),
             encryptionBuffer = key.get_random_key().toBuffer(),
             encryptionKey = passwordAes.encryptToHex(encryptionBuffer),
             privateAesLocal = Aes.fromSeed(encryptionBuffer);
 
-        let walletName = iDB.getCurrentWalletName(),
-            // activeObjKey = KeyGeneratorService.generateActiveKey(brainKey, password),
-            keys = KeyGeneratorService.generateKeys(accountName, password),
-            brainKeyPrivate = keys.active,
-            brainKeyPublic = brainKeyPrivate.toPublicKey().toPublicKeyString(),
-            brainKeyEncrypted = privateAesLocal.encryptToHex(brainKeyPrivate.toBuffer()),
-            passwordPrivate = PrivateKey.fromSeed(password),
-            passwordPublic = passwordPrivate.toPublicKey().toPublicKeyString(),
-            date = new Date(),
-            wallet = {
-                public_name : walletName,
-                password_pubkey : passwordPublic,
-                encryption_key : encryptionKey,
-                encrypted_brainkey : brainKeyEncrypted,
-                brainkey_pubkey : brainKeyPublic,
-                brainkey_sequence: 0,
-                brainkey_backup_date : date,
-                created: date,
-                last_modified: date,
-                chain_id: Apis.instance().chain_id
-            };
+        // The name of the wallet 
+        let walletName = iDB.getCurrentWalletName();
+
+        // Generate the keys using the account name and the password
+        let keys = KeyGeneratorService.generateKeys(account.name, password)
+
+        if (keys.owner.toPublicKey().toPublicKeyString() !== account.owner.key_auths[0][0]) {
```

Comment:
`.first()` vs `[0][0]`:
Any reason to use one or the other?

Reply:
if account is not an immutable then ignore my comment.

Reply:
Account is not immutable. first() is not something we can use.